modules/web-discovery-project/sources/web-discovery-project.es

```diff
@@ -96,1518 +91,27 @@ const WebDiscoveryProject = {
-      -4.673288358097166, -4.633122316371831,
-    ],
-    [
-      -4.806719860900339, -3.6507473420444714, -4.429814962058902,
-      -4.347330952102998, -4.3949170222076255, -4.698366552912107,
-      -4.809448375553543, -4.35423944244681, -4.534137594904308,
-      -4.3611959922401695, -2.871768725634745, -4.598399529122678,
-      -5.564587232141456, -5.519124858064699, -2.4250566837898457,
-      -5.688201188108633, -5.967149580591659, -5.01834957015221,
-      -2.505498094634157, -5.3939617151106924, -6.3813483686685775,
-      -5.360223575478843, -5.821732580733155, -5.742268409378909,
-      -2.5427453130265225, -6.0299504818306895, -5.949907774157153,
-      -6.048642614842842, -6.106911522966818, -6.522426966928483,
-      -1.9368412502165815, -5.384205540165328, -3.2469989725553616,
-      -5.166904264475346, -4.662348253722951, -5.93295821584338,
-      -4.052787789271272, -3.8964417188805776, -4.725629703670472,
-      -3.994277013931218, -3.570836586604418, -4.883136091344702,
-      -4.20182336843176, -4.894970548991704, -4.1342203512338624,
-      -5.159122124033291, -4.442985425248647, -4.740819869164446,
-      -4.583031498894364, -4.589588899440523, -4.4058064220068935,
-      -4.87144005158151, -4.8541485544714496, -4.806719860900339,
-      -4.241282205172307, -4.937530163410501, -4.382360803432213,
-      -5.0116381355642226, -4.368201274828579, -4.782492565565015,
-      -4.544565218066568, -4.4563316672015425,
-    ],
-  ],
-  probHashThreshold: 0.015,
-  probHashChars: {
-    1: 1,
-    0: 0,
-    3: 3,
-    2: 2,
-    5: 5,
-    4: 4,
-    7: 7,
-    6: 6,
-    9: 9,
-    8: 8,
-    A: 36,
-    C: 38,
-    B: 37,
-    E: 40,
-    D: 39,
-    G: 42,
-    F: 41,
-    I: 44,
-    H: 43,
-    K: 46,
-    J: 45,
-    M: 48,
-    L: 47,
-    O: 50,
-    N: 49,
-    Q: 52,
-    P: 51,
-    S: 54,
-    R: 53,
-    U: 56,
-    T: 55,
-    W: 58,
-    V: 57,
-    Y: 60,
-    X: 59,
-    Z: 61,
-    a: 10,
-    c: 12,
-    b: 11,
-    e: 14,
-    d: 13,
-    g: 16,
-    f: 15,
-    i: 18,
-    h: 17,
-    k: 20,
-    j: 19,
-    m: 22,
-    l: 21,
-    o: 24,
-    n: 23,
-    q: 26,
-    p: 25,
-    s: 28,
-    r: 27,
-    u: 30,
-    t: 29,
-    w: 32,
-    v: 31,
-    y: 34,
-    x: 33,
-    z: 35,
-  },
-
+  patterns: new Patterns(),
+  _patternsLastUpdated: null,
   patternsLoader: (() => {
     return new WebDiscoveryProjectPatternsLoader(
       config.settings.ENDPOINT_PATTERNS,
       (content) => {
         try {
-          const { normal, strict } = JSON.parse(content);
-          logger.debug("Got new patterns", { normal, strict });
-          WebDiscoveryProject.contentExtractor.updatePatterns(normal, "normal");
-          WebDiscoveryProject.contentExtractor.updatePatterns(strict, "strict");
-          logger.info("WebDiscoveryProject patterns successfully updated");
+          const rules = JSON.parse(content);
+          logger.debug("Got new patterns", rules);
+          WebDiscoveryProject.patterns.update(rules);
```

Comment:
to confirm, will we no longer have both "normal" and "strict" patterns? what's the reason for this? do we still double fetch? does content extraction only occur with the double fetched page?

Reply:
Yes, the idea is to not have this separation anymore, but extract data from the anonymously fetched page only

Reply:
so we no longer have rules for extracting the query from the page for the purposes of "private query" checking, before the double fetch? we just rely on getting the query via the URL?

Reply:
We currently have some post-processing in place which removes private data from messages when they are extracted. But I agree that it's better to drop suspicious queries before double-fetching. Added this logic here [`fa6331d` (#384)](https://github.com/brave/web-discovery-project/pull/384/commits/fa6331d897487c6ffa296bc2492509e8661e1e24)

Reply:
looks fine, thanks